docs/src/main/asciidoc/includes/tracing/common-callbacks.adoc

```diff
@@ -0,0 +1,159 @@
+
+
+.link:{tracing-javadoc}/Span.Builder.html[`io.helidon.tracing.Span.Builder`] Operations
+|====
+| Method | Purpose | OK?
+| `build()` | Starts the span. | -
+| `end` methods | Ends the span. | -
+| `get()` | Starts the span. | -
+| `kind(Kind)` | Sets the "kind" of span (server, client, internal, etc.) | -
+| `parent(SpanContext)` | Sets the parent of the span to be created from the builder. | -
+| `start()` | Starts the span. | -
+| `start(Instant)` | Starts the span. | -
+| `tag` methods| Add a tag to the builder before the span is built. | &check;
+| `unwrap(Class)` | Cast the builder to the specified implementation type. † | &check;
+|====
+
+† Helidon returns the unwrapped object, not a safe adapter to it.
+
+
+.link:{tracing-javadoc}/Span.html[`io.helidon.tracing.Span`] Operations
+|====
+| Method    | Purpose | OK?
+| `activate()` | Makes the span "current", returning a `Scope`. * | -
+| `addEvent` methods | Associate a string (and optionally other info) with a span. | &check;
+| `baggage()` | Returns the `Baggage` instance associated with the span.| &check;
+| `context()` | Returns the `SpanContext` associated with the span.| &check;
+| `status(Status)` | Sets the status of the span. | -
+| any `tag` method| Add a tag to the span. | &check;
+| `unwrap(Class)` | Cast the span to the specified implementation type. † | &check;
+
+|====
+
+{empty}* Helidon throws the link:{tracing-javadoc}/UnsupportedActivationException.html[`UnsupportedActivationException`] if a listener attempts an illegal operation from inside its `afterActivation` method. This Helidon exception extends `UnsupportedOperationException` and adds the `Scope scope()` method. Callers should catch this exception and close the `Scope`; Helidon will have activated the span and created the scope _before_ it invoked the listeners.
+
+† Helidon returns the unwrapped object, not a safe adapter to it.
+
+.link:{tracing-javadoc}/Scope.html[`io.helidon.tracing.Scope`] Operations
+|====
+| Method    | Purpose | OK?
+| `close()` | Close the scope. | -
+| `isClosed()`| Reports whether the scope is closed. | &check;
+|====
+
+.link:{tracing-javadoc}/SpanContext.html[`io.helidon.tracing.SpanContext`] Operations
+|====
+| Method    | Purpose | OK?
+| `asParent(Span.Builder)` | Sets this context as the parent of a new span builder. | &check;
+| `baggage()` | Returns `Baggage` instance associated with the span context. | &check;
+| `spanId()` | Returns the span ID. | &check;
+| `traceId()` | Returns the trace ID.| &check;
+
+|====
+
+=== Adding a Listener
+==== Explicitly Registering a Listener on a link:{tracing-javadoc}/Tracer.html[`Tracer`]
+Create a `{listener-type}` instance and invoke the `Tracer#register({listener-type})` method to make the listener known to that tracer.
+
+==== Automatically Registering a Listener on all `Tracer` Instances
+
+Helidon also uses Java service loading to locate listeners and register them automatically on all `Tracer` objects.
```

Comment:
I understand we here at Helidon central love services for better or for worse but up until this point I was rolling with my Swing event listeners mental model. Do we want to permit this? I guess probably, given that tracing is so apparently central to what we do?

Reply:
Works for me.